### PR TITLE
git related improvement

### PR DESCRIPTION
**Pull request by xu-cheng**

### CODE REVIEW

File: Library/Homebrew/cmd/doctor.rb [hunk 1]
```diff
@@ -914,12 +906,11 @@ def check_git_newline_settings
   end
 
   def check_git_origin
-    return unless git? && (HOMEBREW_REPOSITORY/".git").exist?
+    return unless Utils.git_available? && (HOMEBREW_REPOSITORY/".git").exist?
```
Comment (mikemcquaid) [inline Library/Homebrew/cmd/doctor.rb]: Just a nit (that you didn't do): I always find `unless ... &&` hard to parse compared to `if ! ...`

**Comment by mikemcquaid** [comment]
:+1:

File: Library/Homebrew/cmd/doctor.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/cmd/doctor.rb]: Fixed and shipped.
